source/iopipe/json/serialize.d

```diff
@@ -560,13 +576,14 @@ OBJ_MEMBER_SWITCH:
                 // any extras should be put in here
                 JSONValue!SType newItem;
                 tokenizer.deserializeImpl(newItem, relPol);
-                __traits(getMember, item, extrasMember).object[name.to!(immutable(SType))] = newItem;
+		auto copyIfNeeded = (string s) => (s.ptr != namebuffer.ptr) ? s : s.idup;
+                __traits(getMember, item, extrasMember).object[copyIfNeeded(name).to!(immutable(SType))] = newItem;
```

Comment:
This change is unnecessary. The name will be copied if it's not a `string`. If it is a `string`, then it's immutable, and should not ever be overwritten.

Someone would have to cast or try really hard to build an immutable string on the stack to make this fail.